lib/chef/resource/archive_file.rb

```diff
@@ -129,6 +129,7 @@ class ArchiveFile < Chef::Resource
             archive.each_entry do |e|
               FileUtils.chown(new_resource.owner, new_resource.group, "#{new_resource.destination}/#{e.pathname}")
             end
+            archive.close
```

Comment:
Looks like [`open_filename`](https://github.com/chef/ffi-libarchive/blob/545d3948e7a78835da48737c1490516905fad2e6/lib/ffi-libarchive/writer.rb#L5) takes a block, and so this would be cleaner and automatically `ensure` closure if these calls were converted to that pattern.